### ISSUE TITLE
[Vertical filter] Truncating and Collapsing groups experience enhancements

### ISSUE THREAD

**Issue by melissaperreault** [Shopify/dawn#1537]
### Observations

Posting here from this comment https://github.com/Shopify/dawn/pull/1443#pullrequestreview-920607437
Issue to discuss further and decide when this can come handy in our roadmap.

**Truncating enhancement** ([Video reference](https://screenshot.click/22-03-219v0-kgm3b.mp4))
- [ ] The [recommendation is to truncate after 10 filters](https://shopify.dev/themes/navigation-search/filtering/storefront-filtering/storefront-filtering-ux#manage-filter-values). Although, clicking a `Show more` link just to display 1 or 2 more filters is not a great experience for buyers. I'd recommend to have a rule. At least doing it to not have to click to reveal 1 filter, I would be good with that. Happy to get your thoughts because it can be arbitrary. cc. @danielvan @nicklepine 
   - If filters = 12 or less don't display `Show more`
   - If filters = 13 or greater display `Show more`

**Collapsing groups**
- [ ] The recommendation is to collapse [after 5 from our internal docs](https://shopify.dev/themes/navigation-search/filtering/storefront-filtering/storefront-filtering-ux#manage-filter-groups).

**Inconsistent filter clearing labels**
- [x] ~Align all instances of the clear/clear all/reset/remove button to `Clear all`, and `[category] [filter name] clear filter` for the individual filter buttons.~ Update: Individual issue created here for quick assistance https://github.com/Shopify/dawn/issues/1547

**Comment by ludoboludo** [comment]
One thing that could potentially be nice is when opening long or many filters, the page's height keeps increasing and you don't get to see the visual updates of the product list being updated. 
We could keep the vertical filters the same height has the product grid and enable scrolling within it: [video comparison with Nike](https://screenshot.click/25-29-njgjc-bcgy3.mp4)

**Comment by bredowmax** [comment]
I know this wasn’t in scope for /vertical-filters but I believe this fits into “experience enhancements”: 

as a user, it is quite annoying when dropdowns host only one checkbox - such as the case with booleans. It results in 2-3x more clicks than necessary. And when dropdowns are open, it makes the vertical filter unnecessarily long

In that case with only one checkbox, I’d rather want to see directly that checkbox, not wrapped inside a collapsible - it would avoid unnecessary clicks and make this vertical filter list much more useable

**Comment by bacounts** [comment]
FILTER option based on a specific product metafield list raised in https://github.com/Shopify/dawn/issues/1481

^^ This could be a better long-term solution or addition especially for the "boonleans experience enhancement" issue raised by @bredowmax in this thread...
> I know this wasn’t in scope for /vertical-filters but I believe this fits into “experience enhancements”:
> 
> as a user, it is quite annoying when dropdowns host only one checkbox - such as the case with booleans. It results in 2-3x more clicks than necessary. And when dropdowns are open, it makes the vertical filter unnecessarily long
> 
> In that case with only one checkbox, I’d rather want to see directly that checkbox, not wrapped inside a collapsible - it would avoid unnecessary clicks and make this vertical filter list much more useable

**Comment by bacounts** [comment]
https://github.com/Shopify/dawn/pull/1443#issuecomment-1081189738

**Comment by melissaperreault** [comment]
### Breakpoint improvements

https://github.com/Shopify/dawn/pull/1443#issuecomment-1077911629

[We can improve the breakpoint for small tablet or large mobile](https://screenshot.click/22-03-ezd6h-ubezg.png). Testing on landscape orientation helps discovering this opportunity.

**Comment by melissaperreault** [comment]
#### Loading overlay

I don't have a complete opinion yet. I have many questions such as the impact on performance depending of the solution we take. From a UX perspective, as buyers are filtering, I don't expect them to scroll further the product grid but there could be sections above and would like to explore a prototype with the recommended implementation before making a final UX call.

**Comment by bacounts** [comment]
On longer Collection pages, increase the usability of the filtering aside by adding a sticky option — you could add a similar feature for the vertical option. The drawer is not needed since it is a popout.

**Comment by martinamarien** [comment]
### Drawer issue (mobile/Desktop)
An additional issue that is already present on main on collection and search: when filters are disabled, the filter options are still present in the drawer. Only the sort options should be visible. 

On Collection, the sort is not visible right now, so I cannot replicate, but I would on Dawn v4.0.0

<details>
<summary>See drawer</summary>
<img width="835" alt="Screen Shot 2022-03-31 at 10 19 18 AM" src="https://user-images.githubusercontent.com/26778637/161077262-25fdb32f-5748-48c3-901e-da0e7422bbaa.png">
</details>

## Collection page
TLDR: UI behaviour is looking pretty good Sofia for the scenarios I tested below. The one critical bug that I think we must fix before shipping is the Sort not showing up when filters are disabled. In my opinion this is critical and currently a broken experience for merchants who are editing their stores. 

### UI improvements

#### Vertical filters

##### Desktop
| **Filter and sort state** | **Filter visible** | **Sort visible** | **Product count visible** | Expected layout |
|---|---|---|---|---|
| Sort enabled only | ✅ No | ✅ Yes | ✅ | ✅ |
| Filters enabled only | ✅ Yes | ✅ No | ⚠️ Not visible (See Desktop 1) | ✅ |
| Filters and sort enabled | ✅ Yes | ✅ Yes | ✅  | ✅ |
| Filters and sort disabled | ✅ No | ✅ No | ❓ Not visible | ✅ |

##### Mobile
| **Filter and sort state** | **Filter visible** | **Sort visible** | **Product count visible** | Expected layout |
|---|---|---|---|---|
| Sort enabled only | ✅ No | ❌ No | ⚠️ Not visible | ❌ ◀️ Sort drawer not present  (See Mobile 1) |
| Filters enabled only | ✅ Yes | ✅ No| ✅ | ✅ Drawer |
| Filters + Sort enabled | ✅ Yes | ✅ Yes | ✅ | ✅ Drawer <br> ⚠️ ◀️  Product count and drawer button misaligned (See Mobile 2) |
| Filters + Sort disabled | ✅ No | ✅ No | ❓Not visible (See Mobile 3) | ✅ |

❌ Critical
❓ Not sure about expected behaviour
◀️ Regression
⚠️ Non-critical
✅ All good

##### Desktop 
- [ ] **1.** When filters are enabled and sort is disabled, the product count is not visible.

    <details>
    <summary>See Image</summary>
  
    <img width="1291" alt="Screen Shot 2022-03-31 at 10 47 07 AM" src="https://user-images.githubusercontent.com/26778637/161083571-c68f54d1-1ab3-4527-aee4-e1538a9bcc5d.png">
    </details>

##### Mobile
- [ ] **1.** When sort is enabled and filtering is disabled, the sort option is not visible, and neither is the product count.

    <details>
    <summary>See image - sort is not available. </summary>
    
    <img width="720" alt="Screen Shot 2022-03-31 at 10 51 37 AM" src="https://user-images.githubusercontent.com/26778637/161084643-071f45d2-fb7d-4bac-b4c2-e5898d755a4c.png">

    </details>

- [ ] **2.** The product result count is misaligned when filter and sort are applied. 

    <details>
    <summary>See image of misaligned text</summary>

    Filter + Sort enabled
    <img width="729" alt="Screen Shot 2022-03-31 at 10 47 42 AM" src="https://user-images.githubusercontent.com/26778637/161083726-10282b72-42ee-4759-a5c5-9c8b51df4bba.png">
    </details>

- [ ] **3.** The product result count is not displayed when the filtering and sort settings are disabled. This is already present on Main (before this was merged). We can flag as a future improvement. 

    <details>
    <summary>See image - product count unavailable </summary>
    
    <img width="707" alt="Screen Shot 2022-03-31 at 10 53 49 AM" src="https://user-images.githubusercontent.com/26778637/161085140-f949a2c5-bb9d-45e1-98b6-23940ceb38f0.png">

    </details>

#### Drawer

##### Desktop
| **Filter and sort state** | **Filter visible** | **Sort visible** | **Product count visible** | Expected layout |
|---|---|---|---|---|
| Sort enabled only | ✅ No | ❌ No | ⚠️ Not visible | ❌ Sort is not available (See desktop 1) |
| Filters enabled only | ✅ Yes | ✅ No | ✅  | ⚠️ ◀️ Product count position is misaligned horizontally (See Desktop 2) |
| Filters and sort enabled | ✅ Yes | ✅ Yes | ✅  | ✅ |
| Filters and sort disabled | ✅ No | ✅ No | ❓ Not visible | ✅ |

##### Mobile
| **Filter and sort state** | **Filter visible** | **Sort visible** | **Product count visible** | Expected layout |
|---|---|---|---|---|
| Sort enabled only | ✅ no | ❌ No | ⚠️ Not visible| ❌ Sort is not available (See Mobile 1) |
| Filters enabled only | ✅ Yes | ✅ No| ✅ | ✅ Drawer |
| Filters + Sort enabled | ✅ Yes | ✅ Yes | ✅ | ✅ Drawer |
| Filters + Sort disabled | ✅ No | ✅ No | ❓ Not visible  | ✅ |

❌ Critical
❓ Not sure about expected behaviour
⚠️ Non-critical
◀️ Regression
✅ All good

##### Desktop 
- [ ] **1.** When sort is enabled and filters is disabled, the sort functionality and product count it not available.

    <details>
    <summary>See image - Sort not available</summary>
   
    <img width="1181" alt="Screen Shot 2022-03-31 at 11 04 23 AM" src="https://user-images.githubusercontent.com/26778637/161087497-66b28c37-cd17-410b-a975-1c1798ca59f6.png">
  </details>

- [ ] **2.** When filters are enabled and sort is disabled, the product count is misaligned horizontally.

    <details>
    <summary>See Image</summary>
  
    <img width="1179" alt="Screen Shot 2022-03-31 at 11 05 19 AM" src="https://user-images.githubusercontent.com/26778637/161087693-b471f22a-b66a-4779-80ba-f356ec2daae5.png">
    </details>

##### Mobile
- [ ] **1.** When sort is enabled and filters is disabled, the sort functionality and product count it not available.

    <details>
    <summary>See image - Sort not available</summary>
   
    <img width="710" alt="Screen Shot 2022-03-31 at 11 08 32 AM" src="https://user-images.githubusercontent.com/26778637/161088357-efd71f9a-a9ea-4bef-8946-65c2cd208ad1.png">
  </details>


#### Horizontal

##### Desktop
| **Filter and sort state** | **Filter visible** | **Sort visible** | **Product count visible** | Expected layout |
|---|---|---|---|---|
| Sort enabled only | ✅ No | ❌ No | ⚠️ Not visible | ❌ Sort is not available (See desktop 1) |
| Filters enabled only | ✅ Yes | ✅ No | ✅  | ✅  |
| Filters and sort enabled | ✅ Yes | ✅ Yes | ✅  | ✅ |
| Filters and sort disabled | ✅ No | ✅ No | ❓ Not visible | ✅ |

##### Mobile
| **Filter and sort state** | **Filter visible** | **Sort visible** | **Product count visible** | Expected layout |
|---|---|---|---|---|
| Sort enabled only | ✅ no | ❌ No | ⚠️ Not visible| ❌ Sort is not available (See Mobile 1) |
| Filters enabled only | ✅ Yes | ✅ No| ✅ | ✅ Drawer |
| Filters + Sort enabled | ✅ Yes | ✅ Yes | ✅ | ✅ Drawer |
| Filters + Sort disabled | ✅ No | ✅ No | ❓ Not visible  | ✅ |

❌ Critical
❓ Not sure about expected behaviour
⚠️ Non-critical
◀️ Regression
✅ All good

##### Desktop 
- [ ] **1.** When sort is enabled and filters is disabled, the sort functionality and product count it not available.

    <details>
    <summary>See image - Sort not available</summary>

    <img width="1180" alt="Screen Shot 2022-03-31 at 11 13 18 AM" src="https://user-images.githubusercontent.com/26778637/161089397-0cf40cb2-8d5d-4bd2-ad92-415064709ca2.png">
  </details>

##### Mobile
- [ ] **1.** When sort is enabled and filters is disabled, the sort functionality and product count it not available.

    <details>
    <summary>See image - Sort not available</summary>
   
    <img width="688" alt="Screen Shot 2022-03-31 at 11 14 56 AM" src="https://user-images.githubusercontent.com/26778637/161089741-1afddd85-7e3d-4f4f-bb7e-4206722a9979.png">


  </details>

**Comment by bredowmax** [comment]
> I know this wasn’t in scope for /vertical-filters but I believe this fits into “experience enhancements”:
> 
> as a user, it is quite annoying when dropdowns host only one checkbox - such as the case with booleans. It results in 2-3x more clicks than necessary. And when dropdowns are open, it makes the vertical filter unnecessarily long
> 
> In that case with only one checkbox, I’d rather want to see directly that checkbox, not wrapped inside a collapsible - it would avoid unnecessary clicks and make this vertical filter list much more useable

Update to this: the Warehouse theme updated this and I love the implementation because it requires much fewer clicks:

Old version that used text metafields with predefined settings:
<img width="341" alt="Screen Shot 2022-04-16 at 10 34 32 AM" src="https://user-images.githubusercontent.com/30573532/163679140-1ddd12b5-dc65-47ed-8f78-cf75fbdfdbd9.png">

New version that uses boolean metafields:
<img width="376" alt="Screen Shot 2022-04-16 at 10 34 17 AM" src="https://user-images.githubusercontent.com/30573532/163679141-da0fb1ff-a5e9-4016-ad96-06e151f5f439.png">
